src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -162,38 +189,65 @@ class InferPythonImportDependencies(InferDependenciesRequest):
     infer_from = PythonSourceField
 
 
+def _get_inferred_resource_deps(
+    all_resource_targets: AllResourcesTargets,
+    detected_resources: ParsedPythonResources,
+) -> Iterator[Address]:
+    resources_by_path: Dict[pathlib.Path, Target] = {}
+    for file_tgt in all_resource_targets.files:
+        resources_by_path[pathlib.Path(file_tgt[FileSourceField].file_path)] = file_tgt
+    for resource_tgt in all_resource_targets.resources:
+        path = pathlib.Path(resource_tgt[ResourceSourceField].file_path)
+        resources_by_path[path] = resource_tgt
+
+    for pkgname, filepath in detected_resources:
+        resource_path = pathlib.Path(*pkgname.split(".")).parent / filepath
+        inferred_resource_tgt = resources_by_path.get(resource_path)
+        if inferred_resource_tgt:
+            yield inferred_resource_tgt.address
+
+
 @rule(desc="Inferring Python dependencies by analyzing imports")
 async def infer_python_dependencies_via_imports(
     request: InferPythonImportDependencies,
     python_infer_subsystem: PythonInferSubsystem,
     python_setup: PythonSetup,
+    all_resource_targets: AllResourcesTargets,
```

Comment:
This is slow to compute, and it's wasted computation if you don't have resource inference enabled. Until we add https://github.com/pantsbuild/pants/issues/12946, the solution is to add `AllResourceAndFileTargetsRequest` and then use `Get(AllResourceAndFileTargets, AllResourceAndFileTargetsRequest())` :/

Reply:
Actually this had made me realize, we might want a union rule for this, and split out the implementation into it's own rule. That also means maybe splitting the resource parsing out of the helper file into its own file.
FWIW this method is called `infer_python_dependencies_via_imports`, which is incorrect for this use case.

Thoughts?